### PR TITLE
dev-util/samurai: Add patch for CVE-2021-30218

### PR DESCRIPTION
**Pull request by orbea**
Also updates to EAPI 8.

Bug: https://bugs.gentoo.org/786951

### CODE REVIEW

**Comment by ajakk** [comment]
Please file security bugs if there's a security issue in one of your packages.

**Comment by ajakk** [comment]
Oh, I'm sorry. This already has a bug:

https://bugs.gentoo.org/786951

In which the old maintainer didn't actually add patches for both issues in the bug.

**Comment by orbea** [comment]
Yea, I failed to notice it needed two patches back then. Additionally I just now noticed the comment for the bug number was wrong...

File: dev-util/samurai/files/samurai-1.2-null_pointer_fix.patch [hunk 1]
```diff
@@ -1,3 +1,35 @@
+CVE-2021-30218 + CVE-2021-30219
```
Comment (thesamesam) [inline dev-util/samurai/files/samurai-1.2-null_pointer_fix.patch:1]: Link to patch?

Comment (orbea) [reply dev-util/samurai/files/samurai-1.2-null_pointer_fix.patch:1]: Force pushed with all the links added.

**Pull request closed without merging**